newIDE/app/src/MainFrame/index.js

```diff
@@ -658,6 +661,29 @@ const MainFrame = (props: Props) => {
 
   useDiscordRichPresence(currentProject);
 
+  const serviceWorkerUpdateStatus = useServiceWorkerUpdateStatus();
+
+  useTimeout(
+    React.useCallback(
+      async () => {
+        if (serviceWorkerUpdateStatus === 'update-ready') {
+          const answer = await showConfirmation({
+            title: t`New update available!`,
+            message: t`A new version of GDevelop is available. Would you like to update now?`,
+          });
+          if (answer) {
+            await updateServiceWorkerToLatestVersion();
```

Comment:
I wonder if we should add a try catch here so that if anything fails, since the user is expecting the page to be reloaded, they can see a (native?) alert that informs them that the update has failed and it should be available the next time they close and reopen GDevelop (if that's even true?)